### PR TITLE
Package updates

### PR DESCRIPTION
**Pull request by lukebakken**

### CODE REVIEW

**Comment by lukebakken** [comment]
@bording thanks! I tried upgrading `System.Memory` via Visual Studio and it wanted to bring in other packages (I forget which) but that doesn't seem to be the case here.

**Comment by bording** [comment]
@lukebakken I was confused for a second when I didn't get a prompt to create a new PR, but I see that you have already! 😄 

I pushed the branch up and then noticed a problem. You know how I was saying it could be risky to bring in packages like System.Memory when using "old" .NET Framework versions? Turns out we were hitting a problem and the new package revealed it.

The older System.Memory package was causing the `net461` target to be referencing the wrong `System.Net.Sockets` assembly, It was referencing a newer version that isn't actually part of .NET Framework 4.6.1 but will be on the system if a newer version of .NET Framework is installed.

That was giving the code access to a `ConnectAsync` overload that it shouldn't have had available, so I had to reintroduce the `#ifdef` around it.

File: projects/RabbitMQ.Client/client/impl/TcpClientAdapter.cs [hunk 1]
```diff
@@ -27,7 +27,11 @@ public virtual async Task ConnectAsync(string host, int port)
                 throw new ArgumentException("No ip address could be resolved for " + host);
             }
 
+#if NETSTANDARD
```
Comment (bording) [inline projects/RabbitMQ.Client/client/impl/TcpClientAdapter.cs]: I could also reverse this and make it be `#if NET461` instead, which would mean any newer version of the framework would automatically use the better API.

This only really matters if the `#ifdef ` is overlooked when the TFM is updated.

**Comment by lukebakken** [comment]
@bording you just answered the question I was about to ask ... which was "how in the heck did these changes expose that build failure???"

File: projects/RabbitMQ.Client/client/impl/TcpClientAdapter.cs (hunk 1 above)
Comment (lukebakken) [reply projects/RabbitMQ.Client/client/impl/TcpClientAdapter.cs]: Choose whatever will be more noticeable in a year when we re-visit this library for `7.0.0`. I'm hoping it won't be that long but you never know...

Comment (bording) [reply projects/RabbitMQ.Client/client/impl/TcpClientAdapter.cs]: I went ahead and pushed up a change.

**Pull request merged**